src/main/java/com/synopsys/integration/detect/lifecycle/boot/DetectBoot.java

```diff
@@ -223,13 +223,11 @@ public DetectBoot(DetectBootFactory detectBootFactory, DetectArgumentState detec
     }
 
     private void publishCollectedPropertyValues(PropertyConfiguration propertyConfiguration, EventSystem eventSystem) throws IllegalAccessException {
-        PropertyInfoCollector propertyInfoCollector = new PropertyInfoCollector(propertyConfiguration);
-        List<PropertyInfo> propertyInfo = propertyInfoCollector.collectPropertyInfo(DetectProperties.allProperties(), PropertyInfoCollector.maskPasswordsAndTokensPredicate());
-        Map<String, String> propertyValues = new HashMap<>();
-        propertyInfo.forEach(
-            it -> propertyValues.put(it.getKey(), it.getValue())
-        );
-        eventSystem.publishEvent(Event.PropertyValuesCollected, new PropertyValues(propertyValues));
+        Map<String, String> rawPropertyKeyValues = propertyConfiguration.getRawKeyValueMap(new HashSet<>(DetectProperties.allProperties()));
+        PropertyMasker propertyMasker = new PropertyMasker();
+        Predicate<String> shouldMaskRawValue = propertyKey -> propertyKey.toLowerCase().contains("password") || propertyKey.toLowerCase().contains("api.token") || propertyKey.toLowerCase().contains("access.token");
+        Map<String, String> maskedRawPropertyKeyValues = propertyMasker.maskRawValues(rawPropertyKeyValues, shouldMaskRawValue);
+        eventSystem.publishEvent(Event.PropertyValuesCollected, new PropertyValues(maskedRawPropertyKeyValues));
```

Comment:
I should maybe have mentioned it before but I am not a fan of this event name either. I think it implies theses are actual values and it would be easy to be confused if you listened for that event and then realized oh these are not the values these are some masked version of them. I don't know that I have a good name but maybe just a thought.

Reply:
Fair.  RawMaskedPropertyValuesCollected?

Reply:
Sounds good.